ttlcache/otel.go

```diff
@@ -59,3 +59,66 @@ func (c *OtelTTLCache[T]) Put(ctx context.Context, id string, value T, expiresAt
 
 	c.cache.Put(ctx, id, value, expiresAt)
 }
+
+func (c *OtelCache[T]) Delete(ctx context.Context, id string) {
+	ctx, span := tracer.Start(ctx, "cache.Delete")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String(ttlCacheNameAttribute, c.name),
+		attribute.String(ttlCacheIDAttribute, id),
+	)
+
+	c.cache.Delete(ctx, id)
+}
+
+// WithPointerOtel wraps a pointer cache and adds OpenTelemetry tracing to it.
+// Since this tracks the id, do not use this if the id is sensitive.
+// This can be safely used with sensitive values.
+func WithPointerOtel[T any](cache PointerCache[T], name string) *PointerOtelCache[T] {
+	return &PointerOtelCache[T]{cache: cache, name: name}
+}
+
+type PointerOtelCache[T any] struct {
+	cache PointerCache[T]
+	name  string
+}
+
+func (c *PointerOtelCache[T]) Get(ctx context.Context, id string, minimumLifetime time.Duration) (*T, bool) {
+	ctx, span := tracer.Start(ctx, "cache.Get")
```

Comment:
does anything distinguish these spans from the otelCache spans / should it? (not sure if "name" will do this, but)

Reply:
No, I would say it's not important to distinguish since only one could be used per name, and names should be unique to avoid clashing metrics

Reply:
And imo the implementation (pointers vs not) is not important to metrics, but it could be derived from the `name` attribute and looking at the code.